controllers/configurationpolicy_controller.go

```diff
@@ -1970,30 +1991,50 @@ func getNamesOfKind(
 	dclient dynamic.Interface,
 	complianceType string,
 	zeroValueEqualsNil bool,
+	existDesiredObj bool,
```

Comment:
8 parameters on this function now 😨

It's worth considering this for a refactor, I think. Not necessarily in this PR, just mentioning it.

Reply:
How about this?  		
`desiredObj *unstructured.Unstructured,`
`if desiredObj != nil {`